### ISSUE TITLE
[a11y]: OverflowMenuItem, None of the descendent elements with "menuitem" role is tabbable

### ISSUE THREAD

**Issue by mateBe95** [carbon-design-system/carbon#12728]
### Package

carbon-components

### Browser

Chrome

### Package version

Newest

### React version

16.12.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker

### Description

<img width="1704" alt="image" src="https://user-images.githubusercontent.com/23103157/203948178-2af16297-e5cb-4fc6-8d10-9e51848620e3.png">

None of the descendent elements with "menuitem" role is tabbable

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&viewMode=story

### Steps to reproduce

https://react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&viewMode=story

1.Open menu.
2. Scan.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@mbgower @tombrunet Is this a valid failure? @francinelucca just addressed the keyboard navigation in https://github.com/carbon-design-system/carbon/pull/12977. OverflowMenuItem navigation happens via arrow keys, not by tabbing.

**Comment by mateBe95** [comment]
Any updates here @tay1orjones the issue still seems to be occurs https://react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&viewMode=story

**Comment by francinelucca** [comment]
> Any updates here @tay1orjones the issue still seems to be occurs https://react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&viewMode=story

In the overflowmenu item navigation is done using the up/down arrow keys. @mbgower bumping, is there something else we should be considering here regarding this violation?

**Comment by mbgower** [comment]
I don't understand the issue. I'm pulling in @tombrunet to have a look.
I conjecture it might be because you're using buttons for the menuitems and putting the negative tabindex on them, and it's setting off a rule. But since they've been given a role of menuitem, I don't see the problem; they're reachable by arrow key.

I _am_ seeing an oddity where when I press tab after opening the menu, the focus is going around the whole list of options. That suggests there's something a little dodgy in the code?

**Comment by tombrunet** [comment]
I need to think about this one a little bit. It's a little weird because as soon as you tab away, it goes away, so the need to tab back isn't really there. The button that launches the menu may need an aria-controls on it to indicate that it's what is launching the menu. But, I'm not sure we handle that situation for this rule yet.
One bit that gives me a little pause is that if you use the mouse, and then click to the browser, the menu is still open. It seems like you can shift tab back to the menu without it disappearing. If the menu went away any time that focus left, it might be a better argument for the tab to not be required.
All of that said, typical menus would have the `menu` be tabbable with active-descendant set to the appropriate menuitem, or would allow the menuitems to be tabbable. It certainly wouldn't hurt anything to do that here, but again, I need to think a little more about if there are any issues in this particular situation if you don't do that.

**Comment by francinelucca** [comment]
> One bit that gives me a little pause is that if you use the mouse, and then click to the browser, the menu is still open. It seems like you can shift tab back to the menu without it disappearing. If the menu went away any time that focus left, it might be a better argument for the tab to not be required. 

Not quite sure how to reproduce this one but we should be able to make the menu close when it loses focus if desired, it might be a bug that it doesn't currently. We can also explore the tabbable-menu option is that is the guidance.

**Comment by tay1orjones** [comment]
Would the new [OverflowMenuV2 implementation](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-overflowmenuv2--overflow-menu-v-2) solve this to some extent? The items are still not tabbable, but it uses the new Menu primitives and might be more properly applying `active-descendant`?

**Comment by francinelucca** [comment]
Synced up with a11y folks, tried adding `aria-controls` to button that controls `ul` visibility and confirmed still experience Accessibility Checker violation. This is potentially a false positive. Further research upcoming on https://github.com/IBMa/equal-access/issues/1402. 

*We still want to add the `aria-controls` attribute to the button on our end.

**Comment by zhanttbj** [comment]
Hi Team, can we have this issue fixed by end of April? thanks

**Comment by francinelucca** [comment]
@zhanttbj It was determined this violation is a false positive from the a11y checker in this case; the fix for that is being tracked https://github.com/IBMa/equal-access/issues/1402. The menu does not use tabs to navigate, it uses up/down arrow. 
We're going to add the `aria-controls` attribute as part of this issue but that won't change the behavior of the component and won't fix the violation until IBMa closes that issue out.

**Comment by janhassel** [comment]
@francinelucca @tombrunet Is `aria-controls` still needed when the relationship between the button and the menu is already established with `aria-owns` (as the `OverflowMenuV2 is doing)`?

**Comment by francinelucca** [comment]
does that mean we should remove the aria-owns property from `OverflowMenuV2` and `MenuButton`? @janhassel @tombrunet 🤔.
I see `OverflowMenuV2` is setting it on the parent div , but `MenuButton` is setting it on the button itself

**Comment by tombrunet** [comment]
If the intent is to have the menu item appear immediately after the button, but you can't do that because of CSS / portal issues, you would have some sort of empty node after the button that owns the menu (e.g., `<div aria-owns="menuId" />`)

**Comment by janhassel** [comment]
@tombrunet @francinelucca 

Good point. The OverflowMenuV2 currently places the `aria-owns` on a container node:
```html
<div aria-owns="menu-id">
  <button>…</button>
</div>

<ul id="menu-id">...</ul>
```

From my manual testing with VoiceOver `aria-owns` worked great for announcement and focus navigation.

I can update `MenuButton` and `ComboButton` as part of #13678 if you agree!

**Comment by francinelucca** [comment]
@tombrunet any input/concern here about Jan's proposal of using `aria-owns` in the button's parent `div`/`span` over using `aria-controls` in the button itself?

**Comment by tombrunet** [comment]
The menu would be owned on the div, but the button still needs aria-controls to indicate what the button is doing, so you need both.

**Comment by francinelucca** [comment]
> The menu would be owned on the div, but the button still needs aria-controls to indicate what the button is doing, so you need both.
got it, thanks!! @tombrunet

**Comment by tay1orjones** [comment]
> The menu would be owned on the div, but the button still needs aria-controls to indicate what the button is doing, so you need both.

@tombrunet This was done in https://github.com/carbon-design-system/carbon/pull/13678 but it still fails `aria_child_tabbable` https://github.com/carbon-design-system/carbon/pull/13678#issuecomment-1572464902

So my understanding (correct me if I'm wrong here) is that this is still blocked by https://github.com/IBMa/equal-access/issues/1402. I'll reopen this for now until that issue is resolved.

**Comment by tw15egan** [comment]
I believe I'm also running into this same issue with `Dropdown` and adding more AVT tests.  You can see the issue by running the a11y checker with the `Dropdown` menu open: https://react.carbondesignsystem.com/iframe.html?args=&id=components-dropdown--default&viewMode=story


Refs https://github.com/carbon-design-system/carbon/issues/14211

**Comment by tombrunet** [comment]
@tay1orjones It looks like our issue was closed shortly after you re-opened this one. Are you still getting this issue?

**Comment by srujana10499** [comment]
Is it still currently blocked? Please let us know if any further details or inputs are required from our side, as this is also affecting our team and we’re keen to help move it forward.
